cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -255,24 +272,91 @@ impl<State> CoreEnvironment<State> {
         let lockfile = Lockfile::read_from_file(&lockfile_path)
             .map_err(CoreEnvironmentError::LockedManifest)?;
 
-        let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
-        pkgdb_cmd.arg("buildenv").arg(lockfile_path);
-
-        let service_config_path = maybe_make_service_config_file(flox, &lockfile)?;
-        if let Some(service_config_path) = &service_config_path {
-            pkgdb_cmd.args(["--service-config", &service_config_path.to_string_lossy()]);
-        }
+        // todo: use `stat` or `nix path-info` to filter out pre-existing store paths
```

Comment:
nit: didn't we decide using stat is a bad idea?

Reply:
Yannik and I discussed that the next step would be to benchmark the performance of the two to decide which is faster in practice, but regarding `stat()` specifically the question was whether the path existed in the store _as a package build is being performed_, and therefore if it was safe to rely upon that.

We concluded the following:

* It is safe with the Linux build sandbox because the path will not exist in the store as the build is being performed, and will only be atomically moved into place at the conclusion of a successful build.
* However, it is _not_ safe on Darwin because its idea of a sandbox doesn't have those same mount namespace capabilities for presenting a path to the build that does not appear in the store. It is doubly unsafe because Darwin doesn't seem to remove the path from the store following a failed build so it's possible for a badly-behaved build to start populating `$out` prior to a failure and leave cruft behind.

I'm personally crossing fingers at this point that `nix path-info` provides a performant way to address the issue, FWIW.